src/structures/interfaces/Collector.js

```diff
@@ -147,6 +147,37 @@ class Collector extends EventEmitter {
     });
   }
 
+  /**
+   * Allows this collector to be consumed with `for await...of`
+   * @type {AsyncIterator}
+   */
+  async * [Symbol.asyncIterator]() {
+    while (!this.ended) {
+      const { value, done } = await new Promise(resolve => {
```

Comment:
instead of copying the content of this.next...
```suggestion
      const { value, done } = await this.next;
```